### PR TITLE
[node] Added `globalThis` support to `global`

### PR DESCRIPTION
**Pull request by JasonHK**
This is a rewrite of #43308 and #43821. Although this PR brings breaking change to `global`, it corrects the type of `global`.

------

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://node.green/#ES2020-features-globalThis>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by JasonHK** [comment]
@jkomyno @a-tarasyuk @alvis @r3nya @btoueg @BrunoScheufler @smac89 @tellnes @Touffy @DeividasBakanas @eyqs @Flarna @Hannes-Magnusson-CK @KSXGitHub @hoo29 @kjin @ajafff @islishude @mwiktorczyk @mohsen1 @n-e @octo-sniffle @galkin @parambirs @eps1lon @SimonSchick @ThomasdenH @WilcoBakker @wwwy3y3 @samuela @kuehlein @j-oliveras @bhongy @chyzwar @trivikr @nguymin4 @yoursunny @qwelias @ExE-Boss @Ryan-Willpower @peterblazejewicz

Could anyone review this PR? I have a feeling that you intentionally ignore my PRs for Node.js...

**Comment by SimonSchick** [comment]
Can we perhaps add these changes to TS3.5 instead to avoid fragmenting versions even further?

**Comment by ExE-Boss** [comment]
You should also add tests for `globalThis` members to `types/node/ts3.4/test/global.ts`.

**Comment by JasonHK** [comment]
> Can we perhaps add these changes to TS3.5 instead to avoid fragmenting versions even further?

@SimonSchick Thanks for your advice, I'll update it today.

**Comment by JasonHK** [comment]
@SimonSchick @ExE-The latest commit should have fixed the issues you mention, please tell me if there are things that also need to fix.

File: types/window-or-global/index.d.ts [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
 // Project: https://github.com/purposeindustries/window-or-global
 // Definitions by: vvakame <https://github.com/vvakame>
+//                 Jason Kwok <https://github.com/JasonHK>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 /// <reference types="node" />
 
 declare var _: Window | typeof global;
+
 export = _;
```
Comment (ExE-Boss) [inline types/window-or-global/index.d.ts:11]:
```suggestion
declare const _: Window | typeof global;
export = _;
```

File: types/window-or-global/ts3.4/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/// <reference types="node" />
+
+declare var _: typeof globalThis & (typeof self | typeof global);
+
+export = _;
```
Comment (ExE-Boss) [inline types/window-or-global/ts3.4/index.d.ts:5]:
```suggestion
declare const _: typeof globalThis & (typeof self | typeof global);
export = _;
```

**Review by ExE-Boss — APPROVED** [review]
**LGTM** with the following nits.

---

**P.S.:** The **Node v12.x** types should use **TS3.4+** instead of **TS3.5+**.

File: types/window-or-global/ts3.4/index.d.ts (hunk 2 above)
Comment (JasonHK) [reply types/window-or-global/ts3.4/index.d.ts]: Updated

File: types/window-or-global/index.d.ts (hunk 1 above)
Comment (JasonHK) [reply types/window-or-global/index.d.ts]: Updated

**Review by ExE-Boss — APPROVED** [review]

File: types/window-or-global/index.d.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
 // Type definitions for window-or-global 1.0
 // Project: https://github.com/purposeindustries/window-or-global
 // Definitions by: vvakame <https://github.com/vvakame>
+//                 Jason Kwok <https://github.com/JasonHK>
```
Comment (ExE-Boss) [inline types/window-or-global/index.d.ts:4]: Arguably, you could add me here as well:
```suggestion
//                 Jason Kwok <https://github.com/JasonHK>
//                 ExE Boss <https://github.com/ExE-Boss>
```

Comment (JasonHK) [reply types/window-or-global/index.d.ts:4]: Updated

**Review by ExE-Boss — APPROVED** [review]

**Pull request merged**